File: Quora/QuestionCorrectionBot/correction_bot.clj
(defn questionCorrectionBot [question] 
   (-> question
       clojure.string/trim
       (str "?")
       (clojure.string/replace  #"," ", ")
       (clojure.string/replace  #"\s+," ",")
       (clojure.string/replace  #"\s+" " ")
       (clojure.string/replace  #"\?+" "?")
       (#(str (clojure.string/capitalize (subs % 0 1)) (subs % 1)))))

File: Quora/QuestionCorrectionBot/CorrectionBot.java
 String questionCorrectionBot(String question){
 question = question
                .trim()
                .concat("?")
                .replace(",",", ")
                .replaceAll("\\s+,",",")
                .replaceAll("\\?+","?")
                .replaceAll("\\s+"," ");
return question.substring(0,1).toUpperCase()+question.substring(1);
}